components/transactions/TransactionRefundButton.js

```diff
@@ -21,65 +22,88 @@ export const refundTransactionMutation = gqlV2/* GraphQL */ `
     setEnabled(false);
     transactionsDataContext?.refetch?.();
   };
 
+  const closeModal = () => {
+    setEnabled(false);
+    setError(null);
+  };
+
+  const rejectButton = () => {
+    return (
+      <StyledButton
+        buttonSize="small"
+        minWidth={140}
+        background="transparent"
+        textTransform="capitalize"
+        onClick={() => setEnabled(true)}
+        disabled={props.disabled}
+      >
+        <FormattedMessage id="transaction.refund.btn" defaultMessage="refund" />
+      </StyledButton>
+    );
+  };
+
   return (
     <Flex flexDirection="column">
       <Box>
-        {!isEnabled && (
-          <StyledButton
-            buttonSize="small"
-            minWidth={140}
-            background="transparent"
-            textTransform="capitalize"
-            onClick={() => setEnabled(true)}
+        {props.disabled ? (
+          <StyledTooltip
+            type="light"
+            content={
+              <FormattedMessage
+                id="transaction.error.thirtydays"
+                defaultMessage="Collective admins cannot refund or reject contributions older than 30 days."
+              />
+            }
           >
-            <FormattedMessage id="transaction.refund.btn" defaultMessage="refund" />
-          </StyledButton>
-        )}
-        {isEnabled && (
-          <React.Fragment>
-            <StyledButton
-              minWidth={140}
-              buttonSize="small"
-              mr={2}
-              textTransform="capitalize"
-              onClick={() => setEnabled(false)}
-            >
-              <FormattedMessage id="form.cancel" defaultMessage="cancel" />
-            </StyledButton>
-            <StyledButton
-              onClick={handleRefundTransaction}
-              buttonSize="small"
-              buttonStyle="dangerSecondary"
-              loading={loading}
-            >
-              <FormattedMessage id="transaction.refund.yes.btn" defaultMessage="Yes, refund!" />
-            </StyledButton>
-          </React.Fragment>
+            {rejectButton()}
+          </StyledTooltip>
+        ) : (
+          rejectButton()
         )}
+        <ConfirmationModal
+          show={isEnabled}
+          onClose={closeModal}
+          cancelLabel={<FormattedMessage id="actions.cancel" defaultMessage="Cancel" />}
+          header={<FormattedMessage id="RefundContribution" defaultMessage="Refund contribution" />}
+          body={
+            <React.Fragment>
+              <FormattedMessage
+                id="transaction.refund.info"
+                defaultMessage="This action will reimburse your contributor for the full amount of their contribution and they will still be a member of your Collective."
+              />
+              {error && <MessageBoxGraphqlError mt="12px" error={error} />}
+            </React.Fragment>
+          }
+          cancelHandler={closeModal}
```

Comment:
The API of this component is not very clear, but you can avoid passing this prop if there's already an `onClose`

```suggestion
```